src/metatensor/models/__init__.py

```diff
@@ -1,8 +1,14 @@
 from pathlib import Path
+import secrets
 
 PACKAGE_ROOT = Path(__file__).parent.resolve()
 
 CONFIG_PATH = PACKAGE_ROOT / "cli" / "conf"
 ARCHITECTURE_CONFIG_PATH = CONFIG_PATH / "architecture"
 
+
+# A constant as "session" variable to set the random seed to a fixed value that do not
+# change within the execution of the program.
+RANDOM_SEED = secrets.randbelow(2**32)
```

Comment:
What does this do? I was wondering if we should set a random seed by default (e.g. 42). This is always useful to guarantee some level of reproducibility even if the user forgets, plus it's very important in distributed training, where each process initializes its own model (and therefore they all need to start from the same seed to have an equivalent copy)

Reply:
It is basically exactly this but the number is not fixed at the whole time but only within one run.

If we would use a fixed number starting a new training run will always lead to same result. This makes averaging of different results harder for a user. To do so you have to set a random number by your own for each run.

Reply:
I like it, as long as it's printed in the log. Here is my log. I didn't find it:

```

[2024-05-14 14:26:53][INFO] - This log is also available in '/home/filippo/code/metatensor-models/tests/resources/outputs/2024-05-14/14-26-53/train.log'.
[2024-05-14 14:26:53][INFO] - Setting up training set
[2024-05-14 14:26:53][WARNING] - Forces not found in section 'energy'. Continue without forces!
[2024-05-14 14:26:53][WARNING] - Stress not found in section 'energy'. Continue without stress!
[2024-05-14 14:26:53][INFO] - Setting up test set
[2024-05-14 14:26:53][INFO] - Setting up validation set
[2024-05-14 14:26:53][INFO] - Setting up model
[2024-05-14 14:26:53][INFO] - Calling architecture trainer
[2024-05-14 14:26:53][INFO] - Checking datasets for consistency
[2024-05-14 14:26:53][INFO] - training on device cuda with dtype torch.float64
[2024-05-14 14:26:53][INFO] - Calculating composition weights
[2024-05-14 14:26:53][INFO] - Setting up data loaders
[2024-05-14 14:26:53][INFO] - Starting training
[2024-05-14 14:26:54][INFO] - Epoch    0, train loss: 3.4033, train energy RMSE: 0.82502, validation loss: 0.64067, validation energy RMSE: 0.56060
[2024-05-14 14:26:56][INFO] - Epoch   10, train loss: 0.0049, train energy RMSE: 0.03141, validation loss: 0.00203, validation energy RMSE: 0.03188
[2024-05-14 14:26:59][INFO] - Epoch   20, train loss: 0.0011, train energy RMSE: 0.01493, validation loss: 0.00165, validation energy RMSE: 0.02989
[2024-05-14 14:27:01][INFO] - Epoch   30, train loss: 0.0008, train energy RMSE: 0.01241, validation loss: 0.00161, validation energy RMSE: 0.02876
[2024-05-14 14:27:03][INFO] - Epoch   40, train loss: 0.0005, train energy RMSE: 0.01004, validation loss: 0.00160, validation energy RMSE: 0.02835
[2024-05-14 14:27:05][INFO] - Epoch   50, train loss: 0.0003, train energy RMSE: 0.00753, validation loss: 0.00153, validation energy RMSE: 0.02557
[2024-05-14 14:27:07][INFO] - Epoch   60, train loss: 0.0002, train energy RMSE: 0.00638, validation loss: 0.00154, validation energy RMSE: 0.02701
[2024-05-14 14:27:09][INFO] - Early stopping criterion reached after 50 epochs without improvement.
[2024-05-14 14:27:09][INFO] - Evaluating training dataset
[2024-05-14 14:27:10][INFO] - energy RMSE: 0.0058243
[2024-05-14 14:27:10][INFO] - Evaluating validation dataset
[2024-05-14 14:27:11][INFO] - energy RMSE: 0.025171
[2024-05-14 14:27:11][INFO] - Evaluating test dataset
[2024-05-14 14:27:12][INFO] - energy RMSE: 0.027529
```

Reply:
I don't print this in the log. But only write this in the `restart_options.yaml`. But it is no problem to also add this to the log file.